packages/common/src/api/tan-query/comments.ts

```diff
@@ -0,0 +1,719 @@
+import {
+  TrackCommentsSortMethodEnum as CommentSortMethod,
+  EntityType
+} from '@audius/sdk'
+import {
+  InfiniteData,
+  useInfiniteQuery,
+  useIsMutating,
+  useMutation,
+  useQuery,
+  useQueryClient
+} from '@tanstack/react-query'
+import { cloneDeep } from 'lodash'
+import { useDispatch } from 'react-redux'
+
+import {
+  commentFromSDK,
+  replyCommentFromSDK,
+  transformAndCleanList
+} from '~/adapters'
+import { useAudiusQueryContext } from '~/audius-query'
+import { Comment, ID, ReplyComment } from '~/models'
+import { toast } from '~/store/ui/toast/slice'
+import { encodeHashId } from '~/utils'
+
+type CommentOrReply = Comment | ReplyComment
+
+const QUERY_KEYS = {
+  trackCommentList: 'trackCommentList',
+  comment: 'comment',
+  replies: 'replies'
+} as const
+
+const messages = {
+  loadError: (type: 'comments' | 'replies') =>
+    `There was an error loading ${type}. Please try again.`,
+  mutationError: (
+    actionType:
+      | 'pinning'
+      | 'unpinning'
+      | 'deleting'
+      | 'posting'
+      | 'editing'
+      | 'reacting to'
+      | 'reporting'
+  ) => `There was an error ${actionType} that comment. Please try again`,
+  muteUserError: 'There was an error muting that user. Please try again.'
+}
+
+/**
+ *
+ * QUERIES
+ *
+ */
+
+type GetCommentsByTrackArgs = {
+  trackId: ID
+  userId: ID | null
+  sortMethod: CommentSortMethod
+  pageSize?: number
+}
+
+export const useGetCommentsByTrackId = ({
+  trackId,
+  userId,
+  sortMethod,
+  pageSize = 5
+}: GetCommentsByTrackArgs) => {
+  const { audiusSdk, reportToSentry } = useAudiusQueryContext()
```

Comment:
Maybe I should rename this a generic `QueryContext`? I feel like we dont need another provider for tan-query

Reply:
Yeah this should probably be "AppContext" or something, this is good for now